spring-amqp/src/main/java/org/springframework/amqp/core/AbstractDeclarable.java

```diff
@@ -103,16 +103,18 @@ public void setIgnoreDeclarationExceptions(boolean ignoreDeclarationExceptions)
 
 	@Override
 	public void setAdminsThatShouldDeclare(Object... adminArgs) {
-		Collection<Object> admins = new ArrayList<>();
-		if (adminArgs != null) {
-			if (adminArgs.length > 1) {
-				Assert.noNullElements(adminArgs, "'admins' cannot contain null elements");
-			}
-			if (adminArgs.length > 0 && !(adminArgs.length == 1 && adminArgs[0] == null)) {
-				admins.addAll(Arrays.asList(adminArgs));
-			}
+		this.declaringAdmins = new ArrayList<>();
```

Comment:
This is not OK. You are breaking an atomicity on the  property and we may and up with a `ConcurrentModificationException` from somewhere.

Another my concern that I don't see a reason in so complex logic here.
The `declaringAdmins` property must be `final` for better performance.

The code of this method should be as simple as:
```
public void setAdminsThatShouldDeclare(Object... adminArgs) {
		Assert.notNull(adminArgs, "'admins' must be provided");
		Assert.noNullElements(adminArgs, "'admins' cannot contain null elements");
		this.declaringAdmins.addAll(Arrays.asList(adminArgs));
}
```

Reply:
Yes, it's more concise but it will break testcase. Will we have any plan to improve codebase in next version? If not, I am not a big fan of `else/else-if condition` and I will try to contribute that reduce `else/else-if condition` for more readability 

https://github.com/spring-projects/spring-amqp/blob/ef23aa9f7f9485ebc50e45a0eb9ed57083c9ca49/spring-rabbit/src/test/java/org/springframework/amqp/rabbit/core/RabbitAdminDeclarationTests.java#L296-L317

Reply:
I see this changed file takes more your energy reviewing, so I will revert it and create a new PR if we have a plan for breaking change in next version.

In this PR, I will try to improve `if-else` without breaking change and `equals` methods for performance in module `spring-amqp`

Reply:
OK. Sounds like a plan.